src/solace_ai_connector/components/inputs_outputs/assembly.py

```diff
@@ -0,0 +1,112 @@
+"""Assembly component for the Solace AI Event Connector"""
+
+from ...common.log import log
+from ..component_base import ComponentBase
+from ...common.message import Message
+
+
+info = {
+    "class_name": "Assembly",
+    "description": (
+        "Assembles messages till criteria is met, "
+        "the output will be the assembled message"
+    ),
+    "config_parameters": [
+        {
+            "name": "assemble_key",
+            "required": True,
+            "description": "The key from input message that would cluster the similar messages together",
+        },
+        {
+            "name": "max_items",
+            "required": False,
+            "default": 10,
+            "description": "Maximum number of messages to assemble. Once this value is reached, the messages would be flushed to the output",
+        },
+        {
+            "name": "max_time_ms",
+            "required": False,
+            "default": 10000,
+            "description": "The timeout in seconds to wait for the messages to assemble. If timeout is reached before the max size is reached, the messages would be flushed to the output",
+        },
+    ],
+    "input_schema": {
+        "type": "object",
+        "properties": {},
+        "required": [],
+    },
+    "output_schema": {
+        "type": "array",
+        "items": {
+            "type": "object",
+            "properties": {
+                "payload": {
+                    "type": "string",
+                },
+                "topic": {
+                    "type": "string",
+                },
+                "user_properties": {
+                    "type": "object",
+                },
+            },
+        },
+    },
+}
+
+# Default timeout to flush the messages
+DEFAULT_FLUSH_TIMEOUT_MS = 10000
+TIMEOUT_PREFIX = "assembly_timeout_"
+
+
+class Assembly(ComponentBase):
+    def __init__(self, **kwargs):
+        super().__init__(info, **kwargs)
+        self.assemble_key = self.get_config("assemble_key")
+        self.max_items = self.get_config("max_items")
+        self.max_time_ms = self.get_config("max_time_ms", DEFAULT_FLUSH_TIMEOUT_MS)
+
+    def invoke(self, message, data):
+        # Check if the message has the assemble key
+        if self.assemble_key not in data:
+            log.error(f"Message does not have the key {self.assemble_key}")
+            raise ValueError(f"Message does not have the key {self.assemble_key}")
+
+        event_key = data[self.assemble_key]
```

Comment:
Do we support all data types for the key? What if this is a dict, array or other object?

Reply:
safest type is to be string as it needs to be hashable, added a type check